File: src/main/java/net/ndrei/teslapoweredthingies/machines/incinerator/IncineratorRecipe.java
package net.ndrei.teslapoweredthingies.machines.incinerator;

import net.minecraft.item.ItemStack;
import net.ndrei.teslapoweredthingies.common.SecondaryOutput;

/**
 * Created by CF on 2017-01-07.
 */
public class IncineratorRecipe {
    public final ItemStack input;
    public final long power;
    public final SecondaryOutput[] secondaryOutputs;

    public IncineratorRecipe(ItemStack input, long power) {
        this(input, power, (SecondaryOutput) null);
    }

    public IncineratorRecipe(ItemStack input, long power, SecondaryOutput secondary) {
        this(input, power, (secondary == null) ? null : new SecondaryOutput[] { secondary });
    }

    public IncineratorRecipe(ItemStack input, long power, SecondaryOutput[] secondaries) {
        this.input = input;
        this.power = power;
        this.secondaryOutputs = secondaries;
    }
}

File: src/main/kotlin/net/ndrei/teslapoweredthingies/machines/incinerator/IncineratorRecipe.kt
package net.ndrei.teslapoweredthingies.machines.incinerator

import net.minecraft.item.ItemStack
import net.minecraft.util.ResourceLocation
import net.ndrei.teslapoweredthingies.MOD_ID
import net.ndrei.teslapoweredthingies.common.BaseTeslaRegistryEntry
import net.ndrei.teslapoweredthingies.common.SecondaryOutput

/**
 * Created by CF on 2017-06-30.
 */
class IncineratorRecipe(name: ResourceLocation, val input: ItemStack, val power: Long, val secondaryOutputs: Array<SecondaryOutput>)
    : BaseTeslaRegistryEntry<IncineratorRecipe>(IncineratorRecipe::class.java, name) {

    constructor(input: ItemStack, power: Long, secondary: SecondaryOutput? = null)
        : this(input, power, if (secondary == null) arrayOf() else arrayOf(secondary))

    constructor(input: ItemStack, power: Long, secondaryOutputs: Array<SecondaryOutput>)
        : this(ResourceLocation(MOD_ID, input.item.registryName.toString().replace(':', '_')), input, power, secondaryOutputs)
}
